File: backend/src/main/java/at/ac/htl/features/ram/RAMMapper.java
package at.ac.htl.features.ram;

import jakarta.enterprise.context.ApplicationScoped;

@ApplicationScoped
public class RAMMapper {
    public RAMDto toResource(RAM ram) {
        return new RAMDto(
                ram.ram_id,
                ram.name,
                ram.price,
                ram.type,
                ram.clock_speed,
                ram.module_count,
                ram.gb_per_module,
                ram.price_per_gb,
                ram.color,
                ram.first_word_latency,
                ram.cas_latency,
                ram.img,
                ram.model
        );
    }
}

File: android_frontend/app/src/main/java/at/htl/leonding/android_frontend/ui/screens/components/mapper/RAMMapper.kt
package at.htl.leonding.android_frontend.ui.screens.components.mapper

import at.htl.leonding.android_frontend.data.model.RamDto
import at.htl.leonding.android_frontend.ui.screens.components.ComponentListItem

fun RamDto.toListItem() = ComponentListItem(
    id = this.id,
    title = this.name,
    subtitle = "${this.moduleCount}x ${this.gbPerModule} GB • ${this.type}-${this.clockSpeed}}",
    imageUrl = this.img
)